proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+
+## Integration Model
+
+* Issue: https://github.com/syndesisio/syndesis-project/issues/22
+* Sprint: 14
+* Affected Repos:
+  - syndesis-rest
+  - syndesis-ui
+
+## Objective
+
+Currently integrations that are deployed are completely disconnected from the Syndesis DB afterwards. There is no way yet to update a running integration nor to see the state of the currently running integration.
+
+The purpose of this document is to suggest a concise state model for `Integrations` as well as define the interaction between the `Integration`s designed in _syndesis-ui_ with the integrations running in the OpenShift cluster whose creation is triggered by a deployment action by the user.
+
+When rethinking the integration and the state model, the following points should be considered:
+
+* Multiple versions of an Integration must be able to exist.
+* Exactly one of these integrations can be active.
+* Each deployment of an Integration creates a new version.
+* Users can switch integration versions from the UI
+* Versioned Integrations can be inspected from a history page.
+* Old integrations can't be changed but can be used as a blueprint for new integrations.
+* As soon as an integration is changed, a new version of an integration is created. All changes go to this new integration until it is deployed by the user. After this, its gets a fixed version and cannot be changed anymore.
+* The user must always be able to see and examine the currently active integration along with its version number.
+
+_Attention: THIS IS WORK-IN-PROGRESS. The propsoal is PRed early to collect early feedback_
+
+
+## Integration state model
+
+Technical an Integration is the domain object describing a Camel Route. 
+It is managed by `syndesis-rest` and `syndesis-ui`. 
+During design time this object only lives in this domain. 
```

Comment:
Question on this one, currently as an integration is created or changed in the UI those changes are only client-side *until* the user decides to actually save the integration.  However the original intent was that changes to the integration would actually be persisted to the backend in some fashion as they're made by the citizen user.

There's pros and cons to each approach, should we figure out which one to use as part of this proposal?